composer.lock

```diff
@@ -1151,30 +1150,30 @@
         },
         {
             "name": "psr/log",
-            "version": "1.1.4",
+            "version": "3.0.0",
```

Comment:
@liayn This is now fixed to `3.0.0`, so you composer *cannot* install it with `php 7.4`.

Reply:
Not sue how you mean this.

The composer.lock file does not mean anything in a composer library package. It is not used, when this project is required by another project.
Best would be to not even have it in version control as it has no benefit. Simply run `composer update` with the PHP version you have locally and things are okay. For CI, when testing multiple PHP versions, you need to `composer update` the packages anyways for each PHP version. Not to speak of "minimal/maximal version"  tests.

Reply:
Not really. In Joomls CMS we have the lock file in the repository, and we run unit and system tests on multiple PHP versions, for Joomla 4 still from PHP 7.2 to 8.2, and that works just fine. See https://github.com/joomla/joomla-cms/blob/4.4-dev/.drone.yml . The lock file is not patched or modified with it, and we use „composer ci“ and not „composer install“ so we can be sure it is not modified.

Reply:
@richard67 Well, no matter how you do it, but you are - of course - fetching the matching dependencies for the environment under test. That is "matching packages for PHP version X". The lock file simply does not matter at this point.

In TYPO3 we have the lock file in the repo too (https://github.com/TYPO3/typo3/blob/main/composer.lock), but for a different reason: We support non-composer based installations (aka "classic"). Hence, upon packaging of the archives, the CI installs the composer packages according to the lock file, ensuring the created archives are idempotent. For composer-based TYPO3 installations, where TYPO3 (more specifically the various packages, e.g. https://github.com/TYPO3/typo3/tree/main/typo3/sysext/core) is a dependency the lock file does not matter at all.
For testing, the used packages are of course modified and the lock files is changed. (Which does not matter, because this only happens in the CI run)
https://github.com/TYPO3/typo3/blob/main/Build/Scripts/runTests.sh#L194

Reply:
> but you are - of course - fetching the matching dependencies for the environment under test.

That's not right. By using always the same lock file and using "npm ci" [npm-ci](https://docs.npmjs.com/cli/v8/commands/npm-ci) instead of "npm install", we always fetch exactly what is in the lock file as resolved version, and so we have the same result on every environment.

Reply:
I'm talking only about PHP and composer and not about the JS world. ;-)

But back to this project: Our code supports all versions of psr/log and we do not need to limit the version artificially. This avoids those nasty situations, where the using project requires some specific version of psr/log and which would potentially clash with what we declare here.
It is totally normal that different versions of some library support different (potentially disjunct) PHP versions.
So, if this project here allows for PHP 8.x it does not matter for us, which version of psr/log is used, as long as it supports the current PHP version and has the same API for us.
Consequently, the version of psr/log, which lands in our lock file, is simply depending on the PHP version you use, when the lock-file is created.

That text got a bit lengthy, but I hope I got my point across, why the lock file does not matter so much for libraries.

Reply:
Ah yes i was confused. Had a long working day, maybe too tired now.

Reply:
Got it now. Sorry for the confusion. :+1: